.github/workflows/ci.yml

```
@@ -1,10 +1,35 @@
 name: CI
 on: [push] 
 jobs: 
- build: 
-  name: Build and test 
-  runs-on: ubuntu-latest 
-  steps: 
-  - uses: actions/checkout@v2 
-  - run: docker build --target test --tag todo-app:test .
-  - run: docker run --env-file .env.test todo-app:test
\ No newline at end of file
+  build: 
+    name: Build and test 
+    runs-on: ubuntu-latest 
+    steps: 
+    - uses: actions/checkout@v2 
+    - run: docker build --target test --tag todo-app:test .
+    - run: docker run --env-file .env.test todo-app:test
+  
+  deploy:
+      name: Deploy 
+      runs-on: ubuntu-latest
+      needs: build
```

Note that while we do want this running for this branch for testing purposes, generally we'll only want it to run on the main branch (we don't want every commit being deployed to production!).
Adding a gateway like below would let us do that
```suggestion
      needs: build
      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```